java/vector/src/main/java/org/apache/arrow/vector/AbstractVariableWidthVector.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import java.nio.ByteBuffer;
+
+import org.apache.arrow.memory.ReusableBuffer;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VariableWidthVector;
+import org.apache.arrow.vector.VectorDefinitionSetter;
+
+/**
+* A base class for common functionalities in variable width vectors.
+*/
+public interface AbstractVariableWidthVector extends VariableWidthVector, FieldVector, VectorDefinitionSetter {
```

Comment:
@lidavidm is this name okay or should we go for something like `VariableWithVectorOperations` or any better suggestion?